File: C++/0227. Basic Calculator II.cpp
class Solution {
public:
    int calculate(string s) {
        vector<int> ss;
        s.push_back('+');
        char presign = '+';
        int temp = 0;

        for(auto c:s){
            if(c == ' ') continue;
            else if(isdigit(c)){
                temp = 10*temp + (c-'0');
            }
            else{
                switch(presign){
                    case '+':
                        ss.push_back(temp);
                        break;
                    case '-':
                        ss.push_back(-temp);
                        break;
                    case '*':
                        ss.back() *= temp;
                        break;
                    default:
                        ss.back() /= temp;
                        break;
                }   
                presign = c;
                temp = 0;
            }
        }
        return accumulate(ss.begin(),ss.end(),0);
    }
};

File: Python3/0227. Basic Calculator II.py
class Solution:
    def calculate(self, s: str) -> int:
        presign = '+'
        s = s+'+'
        ss = []
        temp = 0
        for c in s:
            if c == ' ':
                continue
            elif c.isdigit():
                temp = temp*10+int(c)
            else:
                if presign == '+':
                    ss.append(temp)
                elif presign == '-':
                    ss.append(-temp)
                elif presign == '*':
                    ss[-1] *= temp
                else:
                    ss[-1] = int(ss[-1]/temp)
                temp = 0
                presign = c
        return sum(ss)
